ql/src/java/org/apache/hadoop/hive/ql/exec/tez/HiveSplitGenerator.java

```diff
@@ -385,7 +387,11 @@ public HiveSplitGenerator(InputInitializerContext initializerContext) throws IOE
 
   @Override
   public void onVertexStateUpdated(VertexStateUpdate stateUpdate) {
-    pruner.processVertex(stateUpdate.getVertexName());
+    // pruner registers for vertex state updates after it is ready to handle them
+    // so we do not worry about events coming before pruner was initialized
+    if (pruner != null) {
```

Comment:
I can understand status updates are not needed when not initialized. Is the not null check only for safety or did we introduce any new place where we start assigning null values.

Reply:
Right, this looks like a leftover. I'll remove that check.